KBoard/KBoard/View/MyBoard/AddNewCategory.swift

```diff
@@ -41,9 +61,43 @@ class AddNewCategory: UIViewController, UITextFieldDelegate {
         navigationItem.leftBarButtonItem = UIBarButtonItem.init(title: "Cancel", style: .done, target: self, action: #selector(dismissModal(_:)))
     }
     
-    private func setCategoryNameInput() {
+    private func render() {
         view.addSubview(categoryTextField)
         categoryTextField.anchor(top: view.safeAreaLayoutGuide.topAnchor, left: view.safeAreaLayoutGuide.leftAnchor, right: view.safeAreaLayoutGuide.rightAnchor, paddingTop: 20, paddingLeft: 16, paddingRight: 16)
+        
+        view.addSubview(validationLabel)
+        validationLabel.anchor(top: categoryTextField.bottomAnchor, left: view.safeAreaLayoutGuide.leftAnchor, right: view.safeAreaLayoutGuide.rightAnchor, paddingTop: 20, paddingLeft: 16, paddingRight: 16)
+    }
+    
+    func underLine() {
+        let border = CALayer()
+        border.frame = CGRect(x: 0, y: categoryTextField.frame.size.height+10, width: categoryTextField.frame.width, height: 1)
+        border.borderWidth = 1
+        border.backgroundColor = UIColor.gray.cgColor
+        categoryTextField.layer.addSublayer(border)
+    }
+    
+    @objc private func checkTextValidation(_ notification: Notification) {
+        if let textField = notification.object as? UITextField {
+            if let text = textField.text {
+                
+                let nameSet = Set(categories.map {$0.categoryName})
+                
+                if nameSet.contains(text) == true {
```

Comment:
```suggestion
                if categories.contains { $0.categoryName == text } { }
```

Set 형태로 바꿔줄 필요 없이 바로 `contains` 를 활용해도 좋을 것 같아요!
Set 으로 바꿔주고 탐색한 이유가 있으신가요??!

Reply:
아 이게 제가 어제 카톡에 보낸 오류 상황과 연관이 있는데 ....
아래 코드처럼 array에 contains를 작성하니 "아이돌" 전체 단어를 잡는것이 아니고 "아", "아이", "아이돌" 이렇게 한글자씩 인식을해서..
방법을 찾다가 Set에 contains를 주면 array에 주는것보다 훨씬 시간효율성도 빠르고 단어도 "아이돌"이렇게 전체 값을 잡아준다고 해
Set의 형태로 데이터를 변환해서 넣어준것입니다! ☺️
아래에 이전 코드를 첨부하겠습니다! 보시고 이상한점있으면 말씀부탁드려요!🙃
``` @objc private func checkTextValidation(_ notification: Notification) {

        if let textField = notification.object as? UITextField {
            if let text = textField.text {

                //categories의 1번 인덱스의 categoryName과 텍스트필드 입력값이 일치하면 유효성검사
								//1번 인텍스 categoryName = "아이돌"
                if categories[1].categoryName.contains(text) == true {

                    validationLabel.text = "Category name already exists."
                    validationLabel.textColor = .red

                } else {

                    validationLabel.text = "Good name😎"
                    validationLabel.textColor = .blue

                }
            }
        }
    }

Reply:
아 어제 그게 그 말이었군여,, 그렇다면 すごい입니다..!! 생각을 못했네여🙆
다만 시간효율성을 생각해보면 Set으로 만들어줄때, O(n)타임, nameSet에 contain해준게 O(1)이라서, 결국 똑같을 것 같아요..!

Reply:
오호..!! 그렇군요! 시간효율성이란 개념을 이거 찾다가 처음 접했는데 공부해봐야겠네여..!!